docs/04-api-reference/front-commerce-core/graphql.mdx

```diff
@@ -0,0 +1,435 @@
+- `key` (`string`): key to use to order
+
+_Example_
+
+```ts
+import { reorderForIds } from "@front-commerce/core/graphql";
+
+const orderer = reorderForIds([2, 3, 1], "id");
+const reordered = orderer([{ id: 1 }, { id: 2 }, { id: 3 }]);
+```
+
+## `reorderForIdsCaseInsensitive`
+
+Same as [`reorderForIds`](#) but case insensitive
+
+## `makeBatchLoaderFromSingleFetch`
+
+Convert a single fetch to a batch loader
+
+```ts
+makeBatchLoaderFromSingleFetch(fetch, response);
+```
+
+Argumvents:
+
+- `fetch` (`function`): fetch function
+- `response` (`object`): response transformer
+
+_Example_
+
+```ts
+import { makeBatchLoaderFromSingleFetch } from "@front-commerce/core/graphql";
+
+const axiosInstance = axios.create({
+  baseURL: "https://acme.com/api",
+});
+
+const acmeLoader = makeBatchLoaderFromSingleFetch(
+  (depth) => axiosInstance.get(`/prodcuts`),
+  (response) => response.data.products
+);
+```
+
+## `maybeValue`
+
+Check if a value is defined, else print a debug message and return `null`
+
+```ts
+maybeValue(value, message);
+```
+
+Arguments:
+
+- `value` (`any`): value to check
+- `message` (`string`): debug message to print if value is not defined
+
+_Example_
+
+```ts
+import { maybeValue } from "@front-commerce/core/graphql";
+
+const acmeValue = maybeValue(5, "value is not defined");
+```
+
+## `promiseToMutationSuccess`
+
+Convert a promise to a mutation success object.
+
+```ts
+promiseToMutationSucces(promise);
+```
+
+Arguments:
+
+- `promise` (`Promise`): promise to execute
+
+Returns
+
+- `object`: mutation success
+  - `success`: `boolean`
+  - `data`: `any`
+
+_Example_
+
+```ts
+import { promiseToMutationSuccesData } from "@front-commerce/core/graphql";
+
+const result = await promiseToMutationSuccesData(
+  axios.get("https://acme.com/api/products")
+);
+```
+
+## `promiseToMutationSuccessWithData`
+
+Convert a promise to a mutation success object with data transformer.
+
+```ts
+promiseToMutationSuccesWithData(promise, (transform = (v) => v));
+```
+
+Arguments:
+
+- `promise` (`Promise`): promise to execute
+- `transform` (`function`): transformation function
+
+Returns
+
+- `object`: mutation success
+  - `success`: `boolean`
+  - `data`: `any`
+
+_Example_
+
+```ts
+import { promiseToMutationSuccesWithData } from "@front-commerce/core/graphql";
+
+const result = await promiseToMutationSuccesWithData(
+  axios.get("https://acme.com/api/products"),
+  (data) => ({
+    id: product.id,
+  })
+);
+```
+
+## `withDefault404Result`
+
+Execute a promise and return a default result if the promise is rejected with a
+404 code.
+
+:::tip
+
+This is mainly used for axios request that throw an object with property
+`response.status`. If you want to use it with a promise that isn't an axios
+promise, your promise reject object must include a `response.status` property.
+
+:::
+
+```ts
+withDefault404Result(promise, defaultResult);
+```
+
+Arguments:
+
+- `promise` (`Promise`): promise to execute
```

Comment:
**issue:** "execute" isn't the right term here.

**suggestion:** (I lack the correct formulation, but I think that'll be better).
```suggestion
- `promise` (`Promise`): promise to evaluate
```